screeps-game-api/src/objects/impls/room.rs

```diff
@@ -99,6 +121,86 @@ impl Room {
     //     unimplemented!()
     // }
 
+    pub fn find_path<O, T, F>(
+        &self, 
+        from_pos: &O, 
+        to_pos: &T,
+        opts: FindOptions<F>,
+    ) -> Vec<Step> 
```

Comment:
This only handles non-serialized paths.

Idea:

We could make `FindOptions` generic over "F: FindResultType", a trait like:

```rust
trait FindResultType {
    type Result;

    fn deserialize(v: Value) -> Result;
    fn serialized() -> bool;
}
```

then implement it for `RegularPath`, `SerializedPath` (or names you choose). This would let us return `F::Result` from `find_path` and handle both regular results which are `Vec<Step>`s and serialized results which are `String`s.

Reply:
Wouldn't it be easier just to make a `Path` enum with a `SerializedPath(String)` and a `VectorizedPath(Vec<Step>)` variant?

Reply:
It would be easier, but it then someone who only wants non-serialized paths would have to unwrap the result. Since it's in our power to give type-system-level assurance that the path is a given type, I'd rather do that than require extra runtime checks.

I mean it's fine if we don't handle non-serialized paths for now as long as we don't allow someone to set the 'serialized' FindOptions setting.